### PR TITLE
Add tests for SCIM2 /SChemas endpoint

### PR DESCRIPTION
**Pull request by ayshsandu**
Tests for the fixes in https://github.com/wso2/product-is/issues/10878

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2SchemasTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import java.util.Base64;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.core.Is.is;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.SCHEMAS_ENDPOINT;
+
+public class SCIM2SchemasTest extends SCIM2BaseTest {
+
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2SchemasTest.java:45]:
```suggestion
```

File: modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/claim-management-update-emailverified-claim.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "claimURI": "http://wso2.org/claims/identity/emailVerified",
+  "description": "Email Verified",
+  "displayOrder": 0,
+  "displayName": "Email Verified",
+  "readOnly": false,
+  "regEx": "",
+  "required": false,
+  "supportedByDefault": %b,
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/claim-management-update-emailverified-claim.json:9]: Is the %b intentional?

Comment (ayshsandu) [reply modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/claim-management-update-emailverified-claim.json:9]: Yes, this value is replaced at the test execution time in the test class.

File: modules/integration/tests-integration/tests-backend/src/test/resources/apiSwaggerFiles/scim2.yaml [hunk 3]
```diff
@@ -2411,3 +2438,82 @@ definitions:
...
+      name:
+        type: string
+      mutability:
+        type: string
+      multiValued:
+        type: boolean
+      caseExact:
+        type: boolean
+      returned:
+        type: string
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/resources/apiSwaggerFiles/scim2.yaml:2519]:
```suggestion
        type: string
        
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2SchemasTest.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws IOException, InterruptedException {
+
+        super.testInit(swaggerDefinition, tenant);
+        Thread.sleep(20000);
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2SchemasTest.java:76]: Any specific reason for this?

Comment (ayshsandu) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2SchemasTest.java]: Removed

File: modules/integration/tests-integration/tests-backend/src/test/resources/apiSwaggerFiles/scim2.yaml [hunk 5]
```diff
@@ -1226,7 +1226,35 @@ paths:
           description: Schema not found
           schema:
             $ref: '#/definitions/ErrorSchemaNotFound'
+  /Schemas:
```
Comment (pulasthi7) [inline modules/integration/tests-integration/tests-backend/src/test/resources/apiSwaggerFiles/scim2.yaml:1229]:
```suggestion
  /schemas:
```

**Review by mefarazath — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/resources/apiSwaggerFiles/scim2.yaml (hunk 5 above)
Comment (ayshsandu) [reply modules/integration/tests-integration/tests-backend/src/test/resources/apiSwaggerFiles/scim2.yaml:1229]: This is how the API resource is implemented

**Pull request merged**